File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    
    # 선수들의 초기 랭킹을 딕셔너리 타입으로 저장
    player_ranking = {player:int(rank) for rank,player in enumerate(players)}
    
    for call in callings:
        current_ranking = player_ranking[call]
        
        player_ranking[call] -= 1
        player_ranking[players[current_ranking-1]] += 1
        
        players[current_ranking-1], players[current_ranking] = call, players[current_ranking-1]
        
    return players

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.java
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        String[] answer = {};
        Map<String,Integer> ranking = new HashMap<>();
        
        for (int i = 0;i < players.length;i++) {
            ranking.put(players[i], i);
        }
        
        for (String player : callings) {
            int playerRank = ranking.get(player);
            
            String frontPlayer = players[playerRank - 1];
            
            ranking.replace(frontPlayer, playerRank);
            players[playerRank] = frontPlayer;
            
            ranking.replace(player, playerRank-1);
            players[playerRank-1] = player;
        }
        
        return players;
    }
}